### ISSUE TITLE
Opening find / find+replace while highlighting the document should automatically use 'in selection'

### ISSUE THREAD

**Issue by ProLoser** [atom/find-and-replace#384]
If you press it again, it will change to the document, and hopefully a third time would expand to the project perhaps? Really find this behavior annoying, I'm used to selecting the text automatically trigger "in selection" searching.

**Comment by majelbstoat** [comment]
:+1: I also really miss this behaviour from previous editors.

**Comment by nemoDreamer** [comment]
:-1: I disagree, unless it doesn't conflict with "cmd-e | find-and-replace:use-selection-as-find-pattern".

**Comment by majelbstoat** [comment]
use-selection-as-find-pattern is sticky, which makes it pretty annoying. sometimes I want to search an entire document. sometimes i want to do search and replace within a given function. the sticky nature of the option makes switching between those two modes a pain. cmd-e doesn't solve this use-case.

**Comment by ProLoser** [comment]
@nemoDreamer this would essentially mean that highlighting text on the IDE is an additional way to hit CMD-E and when you de-select the text it returns to the previous state.

**Comment by majelbstoat** [comment]
This does a different thing from cmd-e, right? Cmd-e uses the selected text as the thing to find within the whole document, which is different from searching for something within the selected text.

**Comment by nemoDreamer** [comment]
Correct, @majelbstoat , but if you select a word, then hit `cmd-e` to pre-fill the "find" box, you don't want "in selection" to be automatically activated (because you've got a selection in the editor at the time the "find" panel opens). This ticket suggests that any time you have a selection and open the "find" panel, you'd want to only search within that selection, which is counter to the existing `cmd-e` workflow.

**Comment by majelbstoat** [comment]
Agreed, I think this should only be the case for Cmd-f.

**Comment by ProLoser** [comment]
Actually I don't like that it uses what I have highlighted as my new search
term. I have numerous times done a search, highlighted something while
editing, and continued searching only to realize I'm now searching a
different term. This is not common behavior and has caused horrible
problems for me since you can't easily revert a project wise search and
replace.

I would prefer your selection to be what you search I INSIDE, not the term
to use for searching. At least make this behavior configurable.

**Comment by WalterGR** [comment]
Having "In Selection" be automatically turned on when there's an existing selection is more common than not doing it.

Among programs I have at arm's reach:

Editors / IDEs that do this: IntelliJ IDEA, Emacs, Notepad2, Crimson Editor, PhpStorm, Visual Studio.

Editors / IDEs that do not: Sublime Text, Notepad.exe, Atom.
